internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -458,6 +458,9 @@ func resourceCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.
 		}
 	}
 
+	if err := CheckRegionConfigsPriorityOrder(params.GetReplicationSpecs()); err != nil {
```

Comment:
Why didn't we use validators in the schema instead of in the CRUD operations?

Reply:
oh, you're right that would have been a better choice